esmvalcore/preprocessor/_area.py

```diff
@@ -246,7 +264,7 @@ def area_statistics(cube, operator, fx_files=None):
     if operator == 'mean':
         return cube.collapsed(coord_names,
                               operation,
-                              weights=grid_areas)
+                              weights=np.array(grid_areas))
```

Comment:
Is this explicitly to turn a dask array into a numpy array? Maybe `grid_areas.compute()` or `np.asarray(grid_areas)` would be more appropriate.

Reply:
I can confirm that @zklaus's suggestion to use compute works:

```suggestion
                              weights=grid_areas.compute())
```
Presumably it's better to keep things native in dask than to cast into a numpy array.

Reply:
`.compute()` *does* give you a numpy array. What it *doesn't* do is go through the numpy.array constructor which probably is a bit more performant. If completely sticking with dask arrays is an option, then of course a simple
```
weights=grid_areas)
```
is the best solution.

Reply:
Yep, the stack trace is:
```
2019-07-30 10:35:35,079 UTC [4633] ERROR   Program terminated abnormally, see stack trace below for more information
Traceback (most recent call last):
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_main.py", line 228, in run
    conf = main(args)
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_main.py", line 156, in main
    process_recipe(recipe_file=recipe, config_user=cfg)
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_main.py", line 206, in process_recipe
    recipe.run()
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_recipe.py", line 1126, in run
    self.tasks, max_parallel_tasks=self._cfg['max_parallel_tasks'])
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_task.py", line 594, in run_tasks
    _run_tasks_sequential(tasks)
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_task.py", line 605, in _run_tasks_sequential
    task.run()
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_task.py", line 224, in run
    input_files.extend(task.run())
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/_task.py", line 227, in run
    self.output_files = self._run(input_files)
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/preprocessor/__init__.py", line 394, in _run
    product.apply(step, self.debug)
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/preprocessor/__init__.py", line 261, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/preprocessor/__init__.py", line 206, in preprocess
    result.append(_run_preproc_function(function, item, settings))
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/preprocessor/__init__.py", line 189, in _run_preproc_function
    return function(items, **kwargs)
  File "/users/modellers/ledm/workspace/ESMValTool/ESMValCore/esmvalcore/preprocessor/_area.py", line 267, in area_statistics
    weights=grid_areas) 
  File "/users/modellers/ledm/workspace/ESMValToolTest/miniconda/minicondaLee/envs/esmvaltool2/lib/python3.7/site-packages/iris/cube.py", line 3277, in collapsed
    weights = kwargs["weights"].view()
TypeError: view() missing 1 required positional argument: 'dtype'
```

Bascially, iris can't accept a dask array as a weights field.

Reply:
Ok. I think this can be considered a dask bug and I filed an issue [here](https://github.com/dask/dask/issues/5175). 
Anyway, since we know the weights array to be 2d only, I think it's fine to go with the numpy array. `.compute()` would be my favorite method to get there in this case.

Reply:
Okay, so this line fails when `grid_areas` is still an numpy array (because numpy arrays do have a `compute` method), we need to add a line which casts numpy array `grid_areas` to a dask array.